src/plugins/expressions/common/expression_functions/specs/map_column.ts

```diff
@@ -101,7 +101,14 @@ export const mapColumn: ExpressionFunctionDefinition<
     });
 
     return Promise.all(rowPromises).then((rows) => {
-      const existingColumnIndex = columns.findIndex(({ name }) => name === args.name);
+      const existingColumnIndex = columns.findIndex(({ id, name }) => {
+        // Columns that have IDs are allowed to have duplicate names, for example esaggs
+        if (id) {
```

Comment:
should we also check whether `args.id` is set as well here?

Reply:
@flash1293 @poffdeluxe I've just changed the logic here since your last review. I'm now checking `id === args.name` if the column has an ID, which makes sense to me as a fallback. What do you all think of this change?